features/0337-jwe-compliant-envelope/README.md

```diff
@@ -0,0 +1,239 @@
+
+## Proposed JWE formats
+
+### XC20P content encryption with FIPS compliant ES key wrapping mode (proposed)
+An example format for XC20P encrypted ciphertext with ECDH-ES+A256KW (FIPS compliant key wrapping) would look like this:
+```json
+ {
+  "protected": "eyJlbmMiOiJDMjBQIn0",
+  "recipients": [
+    {
+      "header": {
+        "kid": "urn:123",
+        "alg": "ECDH-ES+A256KW",
+        "epk": {
+          "kty": "OKP",
+          "crv": "X25519",
+          "x": "-3bLMSHYDG3_LVNh-MJvoYs_a2sAEPr4jwFfFjTrmUo"
+        },
+        "apu": "-3bLMSHYDG3_LVNh-MJvoYs_a2sAEPr4jwFfFjTrmUo",
+        "apv": "dXJuOjEyMw"
+      },
+      "encrypted_key": "Sls6zrMW335GJsJe0gJU4x1HYC4TRBZS1kTS1GATEHfH_xGpNbrYLg"
+    }
+  ],
+  "iv": "K0PfgxVxLiW0Dslx",
+  "ciphertext": "Sg",
+  "tag": "PP31yGbQGBz9zgq9kAxhCA"
+}
+```
+where `protected` value is base64Url encoding of this marshaled JSON into bytes:
+```json
+    {
+        "typ": "prs.hyperledger.aries-anon-message",
+        "alg": "ECDH-ES+A256KW",
+        "enc":"XC20P"
+    }
+```
+
+### Fully FIPS compliant content encryption and ES key wrapping mode (JOSE supported - protected `header` proposed)
+The below example is a full FIPS compliant envelope (content encryption and key wrapping):
+```json
+{
+  "protected": "eyJlbmMiOiJBMjU2R0NNIn0",
+  "recipients": [
+    {
+      "header": {
+        "kid": "urn:123",
+        "alg": "ECDH-ES+A256KW",
+        "epk": {
+          "kty": "OKP",
+          "crv": "X25519",
+          "x": "aOH-76BRwkHf0nbGokaBsO6shW9McEs6jqVXaF0GNn4"
+        },
+        "apu": "aOH-76BRwkHf0nbGokaBsO6shW9McEs6jqVXaF0GNn4",
+        "apv": "dXJuOjEyMw"
+      },
+      "encrypted_key": "wXzKi-XXb6fj_KSY5BR5hTUsZIiAQKrxblTo3d50B1KIeFwBR98fzQ"
+    }
+  ],
+  "iv": "9yjR8zvgeQDZFbIS",
+  "ciphertext": "EvIk_Rr6Nd-0PqQ1LGimSqbKyx_qZjGnmt6nBDdCWUcd15yp9GTeYqN_q_FfG7hsO8c",
+  "tag": "9wP3dtNyJERoR7FGBmyF-w"
+}
+```
+where `protected` value is base64Url encoding of this marshaled JSON into bytes:
+```json
+    {
+        "typ": "prs.hyperledger.aries-anon-message",
+        "alg": "ECDH-ES+A256KW",
+        "enc":"A256GCM"
+    }
+```
+
+In the above two examples, `apu` is the encode ephemeral key used to encrypt the cek stored in `encrypted_key` and `apv` is the key id of the static public key of the recipient. Both are base65Url encoded.
+`kid` is the value if an key ID in a DID document that should be resolvable to fetch the raw public key used.
+
+### XC20P content encryption with FIPS compliant SS key wrapping mode (proposed)
+Inspired by (aries-framework-go)[https://github.com/hyperledger/aries-framework-go/tree/v0.1.0/pkg/didcomm/packer/jwe/authcrypt], the proposed format of the envelope will be described as follows:
+```json
+    {
+    "protected": base64url({
+        "typ": "prs.hyperledger.aries-auth-message",
```

Comment:
We should now namespace under didcomm rather than Hyperledger Aries (similar to the message type changes).

Reply:
you mean switch to `didcomm-envelope-enc`?

Reply:
updated to `didcomm-envelope-enc`